base/compiler/tfuncs.jl

```diff
@@ -1554,6 +1558,8 @@ function tuple_tfunc(argtypes::Vector{Any})
                 else
                     params[i] = Type
                 end
+            elseif !isvarargtype(x) && x <: Type
```

Comment:
You need to check hasintersect not subtyping here, then can use Union with Type to widen it successfully

Reply:
I found `hasintersect(x, Type)` here is too permissive as it holds for e.g. `x = Any`.
But I guess it's okay if we do `params[i] = Union{x, Type}`?